contracts/DaoModule.sol

```diff
@@ -65,36 +51,76 @@ abstract contract DaoModule {
     /// @param templateId ID of the template that should be used for proposal questions (see https://github.com/realitio/realitio-dapp#structuring-and-fetching-information)
     /// @notice There need to be at least 60 seconds between end of cooldown and expiration
     constructor(
-        Executor _executor,
+        address _owner,
+        address _executor,
         RealitioV3 _oracle,
         uint32 timeout,
         uint32 cooldown,
         uint32 expiration,
         uint256 bond,
         uint256 templateId
     ) {
+        bytes memory initParams = abi.encode(
+            _owner,
+            _executor,
+            _oracle,
+            timeout,
+            cooldown,
+            expiration,
+            bond,
+            templateId
+        );
+        setUp(initParams);
+    }
+
+    function setUp(bytes memory initParams) public override {
+        (
+            address _owner,
+            address _executor,
+            RealitioV3 _oracle,
+            uint32 timeout,
+            uint32 cooldown,
+            uint32 expiration,
+            uint256 bond,
+            uint256 templateId
+        ) = abi.decode(
+                initParams,
+                (
+                    address,
+                    address,
+                    RealitioV3,
+                    uint32,
+                    uint32,
+                    uint32,
+                    uint256,
+                    uint256
+                )
+            );
+        require(!initialized, "Module is already initialized");
+        require(_executor != address(0), "Executor can not be zero address");
         require(timeout > 0, "Timeout has to be greater 0");
         require(
             expiration == 0 || expiration - cooldown >= 60,
             "There need to be at least 60s between end of cooldown and expiration"
         );
-        executor = _executor;
+        avatar = _executor;
         oracle = _oracle;
         answerExpiration = expiration;
         questionTimeout = timeout;
         questionCooldown = cooldown;
         questionArbitrator = address(_executor);
         minimumBond = bond;
         template = templateId;
-    }
 
-    modifier executorOnly() {
-        require(msg.sender == address(executor), "Not authorized");
-        _;
+        __Ownable_init();
+        transferOwnership(_owner);
+        initialized = true;
```

Comment:
Minor: I would do this right after the require [here](https://github.com/gnosis/dao-module/pull/27/files#diff-726341f5cd357f71eee6420e336880703b7e962cb3b18204e087fbb519f8f8ceR99)